src/app/input/input.component.ts

```diff
@@ -0,0 +1,25 @@
+import { Component, OnInit } from '@angular/core';
+import { RegexService } from '../regex.service';
+
+/**
+ * This component is responsible for taking a regular expression as input and
+ * pushing changes into the observable stream
+ */
+@Component({
+  selector: 'app-input',
+  templateUrl: './input.component.html',
+  styleUrls: ['./input.component.css']
+})
+export class InputComponent implements OnInit {
```

Comment:
I think InputComponent is too general. If we want to add another input component, it would create a problem. What do you thing of renaming this into "SearchInputComponent" or "RegexInputComponent" ? 

After you rename the component, please rename the selector and the filename to be consistent.